iris_grib/tests/unit/grib1_load_rules/test_grib1_load_translations.py

```diff
@@ -0,0 +1,384 @@
+_mock_gribapi.grib_get_double_array = mock.Mock(
+    side_effect=_mock_gribapi_fetch
+)
+_mock_gribapi.grib_is_missing = mock.Mock(
+    side_effect=_mock_gribapi__grib_is_missing
+)
+_mock_gribapi.grib_get_native_type = mock.Mock(
+    side_effect=_mock_gribapi__grib_get_native_type
+)
+
+# define seconds in an hour, for general test usage
+_hour_secs = 3600.0
+
+
+class FakeGribMessage(dict):
+    """
+    A 'fake grib message' object, for testing GribWrapper construction.
+
+    Behaves as a dictionary, containing key-values for message keys.
+
+    """
+
+    def __init__(self, **kwargs):
+        """
+        Create a fake message object.
+
+        General keys can be set/add as required via **kwargs.
+        The keys 'edition' and 'time_code' are specially managed.
+
+        """
+        # Start with a bare dictionary
+        dict.__init__(self)
+        # Extract specially-recognised keys.
+        edition = kwargs.pop("edition", 1)
+        # This testing is only for old-style Grib-1 code.
+        assert edition == 1
+        time_code = kwargs.pop("time_code", None)
+        # Set the minimally required keys.
+        self._init_minimal_message(edition=edition)
+        # Also set a time-code, if given.
+        if time_code is not None:
+            self.set_timeunit_code(time_code)
+        # Finally, add any remaining passed key-values.
+        self.update(**kwargs)
+
+    def _init_minimal_message(self, edition=1):
+        # Set values for all the required keys.
+        # 'edition' controls the edition-specific keys.
+        self.update(
+            {
+                "Ni": 1,
+                "Nj": 1,
+                "numberOfValues": 1,
+                "alternativeRowScanning": 0,
+                "centre": "ecmf",
+                "year": 2007,
+                "month": 3,
+                "day": 23,
+                "hour": 12,
+                "minute": 0,
+                "indicatorOfUnitOfTimeRange": 1,
+                "shapeOfTheEarth": 6,
+                "gridType": "rotated_ll",
+                "angleOfRotation": 0.0,
+                "iDirectionIncrementInDegrees": 0.036,
+                "jDirectionIncrementInDegrees": 0.036,
+                "iScansNegatively": 0,
+                "jScansPositively": 1,
+                "longitudeOfFirstGridPointInDegrees": -5.70,
+                "latitudeOfFirstGridPointInDegrees": -4.452,
+                "jPointsAreConsecutive": 0,
+                "values": np.array([[1.0]]),
+                "indicatorOfParameter": 9999,
+                "parameterNumber": 9999,
+                "startStep": 24,
+                "timeRangeIndicator": 1,
+                "P1": 2,
+                "P2": 0,
+                # time unit - needed AS WELL as 'indicatorOfUnitOfTimeRange'
+                "unitOfTime": 1,
+                "table2Version": 9999,
+            }
+        )
+        # Add edition-dependent settings.
+        self["edition"] = edition
```

Comment:
Arguably you could edit this... but I don't think we need to worry about that right now

Reply:
Actually I think I've misunderstood the purpose of this, so ignore that last comment